rate-limiter/src/token_bucket.rs

```diff
@@ -94,68 +163,70 @@ impl TokenBucket {
 
 #[cfg(test)]
 mod tests {
-    use std::time::{Duration, Instant};
+    use std::{
+        cell::RefCell,
+        time::{Duration, Instant},
+    };
 
     use super::TokenBucket;
 
     #[test]
     fn token_bucket_sanity_check() {
         let limit_per_second = 10;
         let now = Instant::now();
-        let mut rate_limiter = TokenBucket::new_with_now(limit_per_second, now);
+        let time_to_return = RefCell::new(now);
+        let time_provider = || *time_to_return.borrow();
+        let mut rate_limiter = TokenBucket::new_with_now(limit_per_second, now, time_provider);
 
-        assert_eq!(
-            rate_limiter.rate_limit(9, now + Duration::from_secs(1)),
-            None
-        );
+        *time_to_return.borrow_mut() = now + Duration::from_secs(1);
+        assert_eq!(rate_limiter.rate_limit(9), None);
 
-        assert!(rate_limiter
-            .rate_limit(12, now + Duration::from_secs(1))
-            .is_some());
+        *time_to_return.borrow_mut() = now + Duration::from_secs(1);
+        assert!(rate_limiter.rate_limit(12).is_some());
```

Comment:
Can we assert there that the value is exactly Some(duration)?

Reply:
I would leave it this way - it's a basic, kind of black-box sanity check.